pkg/config/config.go

```
@@ -90,12 +90,44 @@ type Config struct {
 	// maps the team name to its config. GitHub doesn't allow duplicated team
 	// names, so we can do safely do this.
 	AllTeams map[string]*TeamConfig `json:"-" yaml:"-"`
+
+	// OverrideTeams is an index of teams in the override file
+	// maps the team name to its config. GitHub doesn't allow duplicated team
+	// names, so we can do safely do this.
+	TeamOverrides map[string]*OverrideTeamConfig `json:"-" yaml:"-"`
+}
+
+// This will hold the information from the Team Override File
+type OverrideConfig struct {
+	// Teams maps the github team name to a OverrideTeamConfig.
+	Teams map[string]*OverrideTeamConfig `json:"teams,omitempty" yaml:"teams,omitempty"`
+}
+
+// OverrideTeamConfig is intended to be made public containing only github usernames and team names
+type OverrideTeamConfig struct {
+	// Members is a list of users that belong to this team.
+	Members []string `json:"members,omitempty" yaml:"members,omitempty"`
+	// Mentors is a list of users that belong to this team that will be excluded from auto review assignments.
+	Mentors []string `json:"mentors,omitempty" yaml:"mentors,omitempty"`
 }
 
 func (c *Config) IndexTeams() {
 	allTeams := map[string]*TeamConfig{}
 	getAllTeams(c.Teams, allTeams)
+	applyTeamOverrides(c.TeamOverrides, allTeams)
 	c.AllTeams = allTeams
+
+}
+
+func applyTeamOverrides(teams map[string]*OverrideTeamConfig, allTeams map[string]*TeamConfig) {
+	for teamName, team := range teams {
+		if _, ok := allTeams[teamName]; ok {
+			allTeams[teamName].Members = team.Members
+			allTeams[teamName].Mentors = team.Mentors
+		} else {
+			fmt.Printf("Override Warning: team %s missing from config\n", teamName)
```

It makes me slightly nervous to see errors just being printed to the commandline and not creating an actual Go error to send up the stack and trigger an actual error in the execution. I suspect this may just be following existing style in this codebase but this means that as a user of the tool, I need to not only run the tool and check that the exit code was successful, but actually scan all the output of the command to understand the operations and whether the changes were in fact successfully applied.

Logging like this is much better than completely ignoring the error conditions, just wondering how severe this error condition is for instance and what the user should end up doing in response to this.

Let's say we take a step back and we end up maybe messing up the aggregation script or someone creates a new team override but something is misspelled, I guess all the tooling will report successful but then in reality the invalid overrides are being completely ignored and the only trace is one line like this in a GitHub workflow somewhere? Would it be better at this stage to return up an error and reject the user input because the overrides are ineffective?

---

I think I was following existing style. But I'll double check that.
I'm happy to turn this into a failure condition.
Being as conservative as possible makes sense.